stacks/api/passport/auth.guard.ts

```diff
@@ -44,16 +44,33 @@ function createAuthGuard(type?: string): Type<CanActivate> {
         strategyType = type;
       }
 
-      const user = await passportFn(strategyType.toLowerCase(), options, (err, user, info) =>
+      let user = await passportFn(strategyType.toLowerCase(), options, (err, user, info) =>
         this.handleRequest(err, user, info, context)
       );
+
+      if (this.isUpdateOwnIdentity(request.method, request.url, user))
+        user = this.attachIdentityAccess(user);
+
       request[options.property || defaultOptions.property] = user;
 
       request.headers["strategy-type"] = strategyType;
 
       return true;
     }
 
+    isUpdateOwnIdentity(method: string, url: string, user: any) {
```

Comment:
We need a more generic and maintainable solution here. ActionGuard can accept a factory function that mutates the policies array on demand.